boa/src/syntax/lexer/string.rs

```diff
@@ -44,9 +44,10 @@ impl StringLiteral {
 
 /// Terminator for the string.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-enum StringTerminator {
+pub(crate) enum StringTerminator {
     SingleQuote,
     DoubleQuote,
+    End,
```

Comment:
What does the `End` refer to? The <code>`</code> character?

Reply:
It refers to the end of the string. `unescape_string` called with `StringTerminator::End` is intended to be used directly for whole strings. In the case of template, it is used to unescape the raw strings, that have already been read until the final `` ` ``.